packages/next/src/server/base-server.ts

```diff
@@ -1093,10 +1093,25 @@ export default abstract class Server<
               parsedUrl.pathname === '/index' ? '/' : parsedUrl.pathname
           }
 
+          // x-matched-path header can be decoded incorrectly
+          // and should only be utf8 characters so this fixes
+          // incorrectly encoded values
+          function fixMojibake(input: string): string {
```

Comment:
can this function be extracted to it's own module with some unit tests?

Reply:
Added